extensions/keycloak-authorization/runtime/src/main/java/io/quarkus/keycloak/pep/runtime/KeycloakPolicyEnforcerAuthorizer.java

```diff
@@ -43,6 +46,11 @@
 
     @Override
     public CheckResult apply(RoutingContext routingContext, SecurityIdentity identity) {
+        if (delegate == null) {
+            LOG.debug(
+                    "Keycloak Policy Enforcer has not been initialized - please make sure 'quarkus.oidc.enabled' is not set to 'false'");
```

Comment:
I think debug logs like this should also be in the `init` method, since it seems that method returns prematurely in this case

Reply:
@gastaldi sure, it is just the way it is currently implemented that a recorder calls the initialization method only if that property is enabled, so the user disabled it, forgot to re-enable and got NPE, so this issue is just about avoiding NPE and saying somethng useful... Pedro @pedroigor is managing this extension, I only venture with some basic fixes into it :-).
Perhaps indeed a deeper refactoring is required here

Reply:
@gastaldi I'll have a quick look as well if it can be caught earlier, there must be a way